aea/protocols/dialogue/base.py

```diff
@@ -948,6 +995,220 @@ def add_dialogue_endstate(
             self._other_initiated[end_state] += 1
 
 
+def find_caller_object(object_type: Type):
+    """Find caller object of certain type in the call stack."""
+    caller_object = None
+    for frame_info in inspect.stack():
+        frame_self = frame_info.frame.f_locals.get("self", None)
+        if not frame_self:
+            continue
+
+        if not isinstance(frame_self, object_type):
+            continue
+        caller_object = frame_self
+    return caller_object
+
+
+class BasicDialoguesStorage:
+    """Dialogues state storage."""
+
+    def __init__(self, dialogues: "Dialogues") -> None:
+        """Init dialogues storage."""
+        self._dialogues_by_dialogue_label = {}  # type: Dict[DialogueLabel, Dialogue]
+        self._dialogue_by_address = defaultdict(
+            list
+        )  # type: Dict[Address, List[Dialogue]]
+        self._incomplete_to_complete_dialogue_labels = (
+            {}
+        )  # type: Dict[DialogueLabel, DialogueLabel]
+        self._dialogues = dialogues
+
+    def setup(self) -> None:
+        """Set up dialogue storage."""
+
+    def teardown(self) -> None:
+        """Tear down dialogue storage."""
+
+    def add(self, dialogue: Dialogue) -> None:
+        """
+        Add dialogue to storage.
+
+        :param dialogue: dialogue to add.
+        :return: None
+        """
+        self._dialogues_by_dialogue_label[dialogue.dialogue_label] = dialogue
+        self._dialogue_by_address[
+            dialogue.dialogue_label.dialogue_opponent_addr
+        ].append(dialogue)
+
+    def remove(self, dialogue_label: DialogueLabel) -> None:
+        """
+        Remove dialogue from storage by it's label.
+
+        :param dialogue_label: label of the dialogue to remove
+        :return: None
+        """
+        dialogue = self._dialogues_by_dialogue_label.pop(dialogue_label, None)
+        if dialogue:
+            self._dialogue_by_address[dialogue_label.dialogue_opponent_addr].remove(
+                dialogue
+            )
+
+    def get(self, dialogue_label: DialogueLabel) -> Optional[Dialogue]:
+        """
+        Get dialogue stored by it's label.
+
+        :param dialogue_label: label of the dialogue
+        :return: dialogue if presents or None
+        """
+        return self.dialogues.get(dialogue_label, None)
+
+    @property
+    def dialogues(self) -> Dict[DialogueLabel, Dialogue]:
```

Comment:
Can we name this please `dialogues_by_dialogue_label` to keep it consistent

Reply:
done